### PR TITLE
Create rating api

### PR DESCRIPTION
**Pull request by zzzzwen**
Issue #8 #9 #10 

Currently not supporting deletion of ratings

### CODE REVIEW

File: backend/controllers/rate.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Load required packages
+var Rate = require('../models/history.js');
```
Comment (Shadowsong27) [inline backend/controllers/rate.js]: line 2: variable naming convention should be lowerCamelCase

File: backend/controllers/rate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+// Create endpoint /api/ratings for POST
+exports.postRates = function (req, res) {
+  var movieId = req.body.movieId;
+  var score = req.body.score;
+  var userId = req.user.id;
+  // Save the rating and check for errors
+  Rate.build({score: score, movie_id: movieId, user_id: userId})
+      .save().then(function (success) {
+        res.json({message: 'Ratings successfully posted!'});
```
Comment (Shadowsong27) [inline backend/controllers/rate.js]: line 12: JSON format, I think break lines around curly brackets should be used, even though it is just one line son

File: backend/controllers/rate.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      });
+};
+
+// Create endpoint /api/ratings for GET
+exports.getRates = function (req, res) {
+  // Use the Ratings model to find all clients
+  Rate.findAll({where: {user_id: req.user.id}}).then(function(ratings){
```
Comment (Shadowsong27) [inline backend/controllers/rate.js]: same above

File: backend/controllers/rate.js [hunk 4]
```diff
@@ -6,17 +6,33 @@ exports.postRates = function (req, res) {
...
+    where: {
+      movie_id: movieId,
+      user_id: userId
+    }
+  }).then(function (rate) {
+    if (rate) {
+      rate.updateAttributes({
+        score: score
+      }).then(function () {
+        return res.json({status: 'success', message: 'Ratings Updated'});
```
Comment (Shadowsong27) [inline backend/controllers/rate.js]: Capitalised every first letter in message may not be needed

**Comment by zzzzwen** [comment]
Updated

**Pull request merged**